### ISSUE TITLE
Throw a meaningfull bug when the openid configuration document is not found

### ISSUE THREAD

**Issue by jmprieur** [AzureAD/microsoft-authentication-library-for-dotnet#883]
**Which Version of MSAL are you using ?**
MSAL.NET 3.0

**Platform**
Any (console app below)

**Is this a new or existing app?**
Existing B2C app moving to a sovereign cloud (maybe not well configured)

**Repro**

```csharp
using Microsoft.Identity.Client;

namespace sovereign_china
{
    class Program
    {
        public static PublicClientApplication pca = null;

        // Azure AD B2C Coordinates
        public static string Tenant = "name.partner.onmschina.cn";
        public static string AzureADB2CHostname = "login.chinacloudapi.cn";
        public static string ClientID = "<appId>";
        public static string PolicySignUpSignIn = "b2c_1_siupin";

        public static string[] Scopes = { "https://fabrikamb2c.onmicrosoft.com/helloapi/demo.read" };
        public static string ApiEndpoint = "https://fabrikamb2chello.azurewebsites.net/hello";

        public static string AuthorityBase = $"https://{AzureADB2CHostname}/tfp/{Tenant}/";
        public static string Authority = $"{AuthorityBase}{PolicySignUpSignIn}";

        // default redirectURI; each platform specific project will have to override it with its own
        static void Main(string[] args)
        {
            pca = new PublicClientApplication(ClientID, Authority);
            pca.AcquireTokenAsync(Scopes).GetAwaiter().GetResult() ;
        }
    }
}
```

**Expected behavior**
1. Ideally, we get a token
2. but since the openid connect description document (https://login.chinacloudapi.cn/tfp/name.partner.onmschina.cn/b2c_1_siupin/v2.0/.well-known/openid-configuration) for the authority host does not exist, throw a meaningful `MsalServiceException` 

**Actual behavior**
a System.NullReferenceException is thrown.

```Text
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Microsoft.Identity.Client
  StackTrace:
   at Microsoft.Identity.Client.OAuth2.OAuth2Client.<ExecuteRequestAsync>d__10`1.MoveNext() in C:\gh\msal.net\src\Microsoft.Identity.Client\OAuth2\OAuth2Client.cs:line 141
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Identity.Client.Instance.AuthorityEndpointResolutionManager.<DiscoverEndpointsAsync>d__6.MoveNext() in C:\gh\msal.net\src\Microsoft.Identity.Client\Instance\AuthorityEndpointResolutionManager.cs:line 175
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Identity.Client.Instance.AuthorityEndpointResolutionManager.<ResolveEndpointsAsync>d__3.MoveNext() in C:\gh\msal.net\src\Microsoft.Identity.Client\Instance\AuthorityEndpointResolutionManager.cs:line 92
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<ResolveAuthorityEndpointsAsync>d__19.MoveNext() in C:\gh\msal.net\src\Microsoft.Identity.Client\Internal\Requests\RequestBase.cs:line 300
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.<ExecuteAsync>d__8.MoveNext() in C:\gh\msal.net\src\Microsoft.Identity.Client\Internal\Requests\InteractiveRequest.cs:line 89
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<RunAsync>d__13.MoveNext() in C:\gh\msal.net\src\Microsoft.Identity.Client\Internal\Requests\RequestBase.cs:line 161
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Identity.Client.PublicClientApplication.<Microsoft-Identity-Client-ApiConfig-IPublicClientApplicationExecutor-ExecuteAsync>d__39.MoveNext() in C:\gh\msal.net\src\Microsoft.Identity.Client\PublicClientApplication.WithBuilders.cs:line 182
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Identity.Client.PublicClientApplication.<AcquireTokenAsync>d__3.MoveNext() in C:\gh\msal.net\src\Microsoft.Identity.Client\PublicClientApplication.cs:line 123
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at sovereign_china.Program.Main(String[] args) in C:\Users\jmprieur\source\repos\local\sovereign-china\sovereign-china\Program.cs:line 25
```

**Comment by jmprieur** [comment]
@henrik-me, as discussed.

**Comment by jennyf19** [comment]
fixed in [2.7.1 release](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/2.7.1)
